src/Symfony/Bundle/FrameworkBundle/Routing/DelegatingLoader.php

```diff
@@ -46,16 +48,23 @@ public function __construct(ControllerNameParser $parser, LoggerInterface $logge
     }
 
     /**
-     * Loads a resource.
-     *
-     * @param mixed  $resource A resource
-     * @param string $type     The resource type
-     *
-     * @return RouteCollection A RouteCollection instance
+     * {@inheritdoc}
      */
     public function load($resource, $type = null)
     {
-        $collection = parent::load($resource, $type);
+        if (false !== $this->loading) {
+            // This can happen if a fatal error occurs in parent::load()
```

Comment:
But this will now only work with this routing loader. If the user uses a different loader, he has the same problem. So I feel this must be handled generically on top of the loading, e.g. with a decorator.